IPython/lib/display.py

```diff
@@ -383,36 +383,41 @@ def __init__(self,
         result_html_suffix : str
             text to append at the end of link [default: '<br>']
         """
-        if isdir(path):
+        if Path(path).is_dir():
             raise ValueError("Cannot display a directory using FileLink. "
               "Use FileLinks to display '%s'." % path)
-        self.path = fsdecode(path)
+        self.path = Path(path)
```

Comment:
@sumagnadas could you try `self.path = Path(fsdecode(path)) instead?

Reply:
That can be done.